src/staking/observable/index.ts

```diff
@@ -0,0 +1,372 @@
+import { networks, Psbt, Transaction } from "bitcoinjs-lib";
+import { ObservableStakingParams } from "../../types/params";
+import { UTXO } from "../../types/UTXO";
+import { StakingScripts } from "../stakingScript";
+import { StakingScriptData } from "../stakingScript";
+import { StakingError, StakingErrorCode } from "../../error";
+import { 
+  stakingTransaction, unbondingTransaction,
+  withdrawEarlyUnbondedTransaction,
+  withdrawTimelockUnbondedTransaction
+} from "..";
+import { 
+  getPublicKeyNoCoord, isTaproot,
+  isValidBitcoinAddress, isValidNoCordPublicKey
+} from "../../utils/btc";
+import { getStakingTerm, validateParams, validateStakingTxInputData } from "../../utils/staking";
+import { PsbtTransactionResult } from "../../types/transaction";
+
+interface StakerInfo {
+  address: string;
+  publicKeyNoCoordHex: string;
+}
+
+interface ObservableStakingTransaction {
+  txHex: string;
+  outputIndex: number;
+  startHeight: number;
+  timelock: number;
+}
+
+interface ObservableDelegation {
```

Comment:
what is conceptual difference between ` ObservableStakingTransaction` and `ObservableDelegation` ? Asking as I try to fine reasoning why some fields are in one and other in other. 

For example, one can argue that ` stakingTxHashHex` is more of the propery of ` ObservableStakingTransaction` rather than `ObservableDelegation`